File: BECore/app/model/Employee.js
/// <summary>
/// Model EXTJS Employee.
/// </summary>
Ext.define('QLDT.model.Employee', {
    extend: 'Ext.data.Model',

    idProperty: 'EmployeeID',
    idType: 'Guid',

    code: 'EmployeeCode',
    name: 'FullName',

    fields: [
        { name: "EmployeeID", type: "string" },
		{ name: "CompanyID", type: "string" },
		{ name: "EmployeeCode", type: "string" },
		{ name: "Password", type: "string" },
        { name: "IsChangedPassword", type: "boolean" },
		{ name: "FirstName", type: "string" },
		{ name: "LastName", type: "string" },
		{ name: "FullName", type: "string" },
		{ name: "RankID", type: "string" },
		{ name: "Gender", type: "int" },
		{ name: "AcademicRankID", type: "string" },
		{ name: "YearOfAcademicRank", type: "int" },
		{ name: "DegreeID", type: "string" },
		{ name: "YearOfDegree", type: "int" },
		{ name: "PositionID", type: "string" },
		{ name: "BirthDay", type: "date" },
		{ name: "BirthPlace", type: "string" },
		{ name: "HomeLand", type: "string" },
		{ name: "NativeAddress", type: "string" },
		{ name: "Tel", type: "string" },
		{ name: "HomeTel", type: "string" },
		{ name: "Mobile", type: "string" },
		{ name: "Fax", type: "string" },
		{ name: "Email", type: "string" },
		{ name: "OfficeAddress", type: "string" },
		{ name: "HomeAddress", type: "string" },
		{ name: "Website", type: "string" },
		{ name: "Description", type: "string" },
		{ name: "FileResourceID", type: "string" },
		{ name: "Inactive", type: "boolean" },
		{ name: "SortOrder", type: "int" },
        { name: "IDNumber", type: "string" },
        { name: "IssuedBy", type: "string" },
        { name: "DateBy", type: "date" },
        { name: "AccountNumber", type: "string" },
        { name: "Bank", type: "string" },

    ]
});

File: MT.Model/Dictionary/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MT.Model
{
    #region Employee:EntityBase
    /// <summary>
    /// This object represents the properties and methods of a Employee:EntityBase.
    /// </summary>
    public class Employee : EntityBase
    {
        [Key]
        public Guid EmployeeID { get; set; }
        public Guid CompanyID { get; set; }
        public string EmployeeCode { get; set; }
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName { get; set; }
        public Guid RankID { get; set; }
        public int Gender { get; set; }

        public string GenderName
        {
            get
            {
                string name = "Nam";
                if (this.Gender == 1)
                {
                    name = "Nữ";
                }
                return name;
            }
        }
        public Guid AcademicRankID { get; set; }
        public int YearOfAcademicRank { get; set; }
        public Guid DegreeID { get; set; }
        public int YearOfDegree { get; set; }
        public Guid PositionID { get; set; }
        public DateTime BirthDay { get; set; }
        public string BirthPlace { get; set; }
        public string HomeLand { get; set; }
        public string NativeAddress { get; set; }
        public string Tel { get; set; }
        public string HomeTel { get; set; }
        public string Mobile { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string OfficeAddress { get; set; }
        public string HomeAddress { get; set; }
        public string Website { get; set; }
        public string Description { get; set; }
        public string FileResourceID { get; set; }
        public bool Inactive { get; set; }
        public int SortOrder { get; set; }
        public string IDNumber { get; set; }
        public string IssuedBy { get; set; }
        public DateTime DateBy { get; set; }
        public string AccountNumber { get; set; }
        public string Bank { get; set; }
        public bool IsChangedPassword { get; set; }

    }
    #endregion
}

